src/auto_scheduler/loop_state.h

```diff
@@ -225,6 +238,16 @@ class StateNode : public Object {
    * operation.
    */
   AttachMap attach_map;
+  /*!
+   * \brief The up-to-date ComputeDAG of this state, used for some steps that may change the
+   * stage structure of the ComputeDAG (e.g. CacheReadStep/CacheWriteStep). This will alway be kept
+   * up-to-date, while the original ComputeDAG may not be up-to-date.
+   * The default value is an empty NullOpt, means no modification to the original DAG.
+   * Typical usage for this is when acquiring information from ComputeDAG (e.g. check for its
+   * AccessAnalyzer), use the `current_compute_dag` first, if it's Null, use the original
+   * ComputeDAG.
+   */
```

Comment:
```suggestion
   * \brief The up-to-date ComputeDAG of this state. If the value is null, it means no steps modify the DAG. 
   * The dag is the same as the compute_dag in the `SearchTask`. Otherwise, it means some steps 
   * (e.g., CacheReadStep/CacheWriteStep) modify the dag, then the stored value is the new and up-to-date
   * DAG for this state.
   */
```